syntax = "proto3";

message Session {
    enum Type {
        NONE = 0;
        REQUEST1 = 1;
        RESPONSE1 = 2;
        REQUEST2 = 3;
        RESPONSE2 = 4;
        OPERATOR1 = 11;
        OPERATOR2 = 12;
        OPERATOR3 = 13;
        OPERATOR4 = 14;
        OPERATOR5 = 15;
        OPERATOR6 = 16;
    }
}

message OpDef {
    string name = 1;
    repeated Session.Type inputs = 2;
    Session.Type output = 3;
}

message GraphDef {
    string name = 1;
    repeated OpDef op_defs = 2;
}

message GraphsConf {
    repeated GraphDef graph_defs = 1;
}


